### PR TITLE
feat(CHZZK): add presence

### PR DESCRIPTION
**Pull request by Slowlife01**
## Description 

User want it to be a separate Presence, they don't want to show other Naver services

## Acknowledgements
- [x] I read the [Presence Guidelines](https://github.com/PreMiD/Presences/blob/main/.github/CONTRIBUTING.md)
- [x] I linted the code by running `yarn format`
- [x] The PR title follows the repo's [commit conventions](https://github.com/PreMiD/Presences/blob/main/.github/COMMIT_CONVENTION.md)

## Screenshots
<details>
<summary> Proof showing the creation/modification is working as expected </summary>

image
image
image
image

</details>

### CODE REVIEW

File: websites/C/CHZZK/presence.ts [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+
+const enum Assets {
+	Logo = "https://cdn.discordapp.com/app-assets/1232944311415603281/1232944461047267338.png?size=512",
+}
+
+const enum ChzzkAssets {
+	Browse = "https://cdn.discordapp.com/app-assets/1232944311415603281/1232952455562526731.png",
+	Live = "https://cdn.discordapp.com/app-assets/1232944311415603281/1232952455369850910.png",
+	Play = "https://cdn.discordapp.com/app-assets/1232944311415603281/1232952455625707581.png",
+	Pause = "https://cdn.discordapp.com/app-assets/1232944311415603281/1232952455101419540.png",
```
Comment (theusaf) [inline websites/C/CHZZK/presence.ts:30]: Assets should be 512x512px. You can also consider using the default assets.
```suggestion
	Browse = "https://cdn.discordapp.com/app-assets/1232944311415603281/1232952455562526731.png?size=512",
	Live = "https://cdn.discordapp.com/app-assets/1232944311415603281/1232952455369850910.png?size=512",
	Play = "https://cdn.discordapp.com/app-assets/1232944311415603281/1232952455625707581.png?size=512",
	Pause = "https://cdn.discordapp.com/app-assets/1232944311415603281/1232952455101419540.png?size=512",
```

File: websites/C/CHZZK/presence.ts [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+						"h2[class^=video_information_title]"
+					);
+					presenceData.state = document.querySelector(
+						"p[class^=video_information_name]"
+					);
+					presenceData.largeImageKey = (await presence.getSetting("logo"))
+						? streamerLogo.href.replace(streamerLogo.search, "")
+						: Assets.Logo;
+
+					if (location.pathname.startsWith("/live")) {
```
Comment (theusaf) [inline websites/C/CHZZK/presence.ts:77]:
```suggestion
					if (document.location.pathname.startsWith("/live")) {
```

File: websites/C/CHZZK/presence.ts [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+						presenceData.smallImageKey = ChzzkAssets.Live;
+						presenceData.smallImageText = strings.live;
+						presenceData.startTimestamp =
+							Math.floor(Date.now() / 1000) -
+							presence.timestampFromFormat(
+								document.querySelector("span[class^=video_information_count]")
+									.textContent
+							);
+						presenceData.buttons = [
+							{ url: location.href, label: strings.watchStream },
```
Comment (theusaf) [inline websites/C/CHZZK/presence.ts:87]:
```suggestion
							{ url: document.location.href, label: strings.watchStream },
```

File: websites/C/CHZZK/presence.ts [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+						];
+					} else {
+						presenceData.smallImageKey = ChzzkAssets.Play;
+						presenceData.smallImageText = strings.play;
+						[presenceData.startTimestamp, presenceData.endTimestamp] =
+							presence.getTimestampsfromMedia(video);
+						presenceData.buttons = [
+							{ url: location.href, label: strings.watchVideo },
```
Comment (theusaf) [inline websites/C/CHZZK/presence.ts:95]:
```suggestion
							{ url: document.location.href, label: strings.watchVideo },
```

**Review by theusaf — CHANGES_REQUESTED** [review]

File: websites/C/CHZZK/presence.ts [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+	Play = "https://cdn.discordapp.com/app-assets/1232944311415603281/1232952455625707581.png?size=512",
+	Pause = "https://cdn.discordapp.com/app-assets/1232944311415603281/1232952455101419540.png?size=512",
+}
+
+let oldLang: string, strings: Awaited<ReturnType<typeof getStrings>>;
+
+presence.on("UpdateData", async () => {
+	const newLang = await presence.getSetting<string>("lang"),
+		showStreamerLogo = await presence.getSetting<boolean>("logo"),
+		showElapsedTime = await presence.getSetting<boolean>("time");
```
Comment (theusaf) [inline websites/C/CHZZK/presence.ts:38]: For improved performance, use `Promise.all`

**Review by theusaf — CHANGES_REQUESTED** [review]

**Review by Timeraa — APPROVED** [review]

**Review by theusaf — APPROVED** [review]

**Pull request merged**